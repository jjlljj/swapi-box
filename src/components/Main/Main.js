import React from 'react';
import { arrayOf, func, object} from 'prop-types';
import './Main.css';
import { Switch, Route } from 'react-router-dom';
import Welcome from '../Welcome/Welcome';
import Favorites from '../Favorites/Favorites';
import Planets from '../Planets/Planets';
import People from '../People/People';
import Vehicles from '../Vehicles/Vehicles';

const Main = ({ 
  fetchPeople, 
  fetchPlanets, 
  fetchVehicles, 
  addToFav, 
  people, 
  planets, 
  vehicles, 
  favorites, 
  openingText 
}) => {

  return (
    <div> 
      <Switch>
        
        <Route path='/people' 
          render={() => (
            <People 
              fetchData={fetchPeople} 
              addToFav={addToFav}
              cards={people} />
          )}
        />

        <Route path='/planets' activeClassName="active"
          render={() => (
            <Planets 
              fetchData={fetchPlanets} 
              addToFav={addToFav}
              cards={planets} />
          )}
        />

        <Route path='/vehicles' 
          render={() => (
            <Vehicles 
              fetchData={fetchVehicles}       
              cards={vehicles} /> 
          )}
        />

        <Route path='/favorites' 
          render={() => (
            <Favorites 
              addToFav={addToFav}
              cards={favorites} /> 
          )}
        />

        <Route path='/' 
          render={() => (
            <Welcome {...openingText} />
          )}
        />
        
      </Switch>
    </div>
  );
};

Main.propTypes = {
  fetchPeople: func, 
  fetchPlanets: func, 
  fetchVehicles: func, 
  addToFav: func, 
  people: arrayOf(object), 
  planets: arrayOf(object), 
  vehicles: arrayOf(object), 
  favorites: arrayOf(object), 
  openingText: object
};

export default Main;
