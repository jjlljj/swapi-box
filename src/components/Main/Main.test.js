/* eslint-disable */

import React from 'react';
import ReactDom from 'react-dom';
import { shallow } from 'enzyme';
import Main from './Main';

describe('Main', () => {

  it('should exist', () => {
    const renderedComponent = shallow(<Main />);

    expect(renderedComponent).toBeDefined();
  })

})
