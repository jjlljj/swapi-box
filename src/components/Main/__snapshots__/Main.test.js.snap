// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Switch>
        <Route
                path="/people"
                render={[Function]}
        />
        <Route
                activeClassName="active"
                path="/planets"
                render={[Function]}
        />
        <Route
                path="/vehicles"
                render={[Function]}
        />
        <Route
                path="/favorites"
                render={[Function]}
        />
        <Route
                path="/"
                render={[Function]}
        />
</Switch>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            path="/people"
            render={[Function]}
/>,
          <Route
            activeClassName="active"
            path="/planets"
            render={[Function]}
/>,
          <Route
            path="/vehicles"
            render={[Function]}
/>,
          <Route
            path="/favorites"
            render={[Function]}
/>,
          <Route
            path="/"
            render={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/people",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeClassName": "active",
            "path": "/planets",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/vehicles",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/favorites",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
                    path="/people"
                    render={[Function]}
          />
          <Route
                    activeClassName="active"
                    path="/planets"
                    render={[Function]}
          />
          <Route
                    path="/vehicles"
                    render={[Function]}
          />
          <Route
                    path="/favorites"
                    render={[Function]}
          />
          <Route
                    path="/"
                    render={[Function]}
          />
</Switch>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              path="/people"
              render={[Function]}
/>,
            <Route
              activeClassName="active"
              path="/planets"
              render={[Function]}
/>,
            <Route
              path="/vehicles"
              render={[Function]}
/>,
            <Route
              path="/favorites"
              render={[Function]}
/>,
            <Route
              path="/"
              render={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/people",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeClassName": "active",
              "path": "/planets",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/vehicles",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/favorites",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
